operations/mimir-tests/test-disable-consul-enable-memberlist.jsonnet

```diff
@@ -0,0 +1,14 @@
+local mimir = import 'mimir/mimir.libsonnet';
+
+mimir {
```

Comment:
can we base this jsonnet on test-gossip.jsonnet (copy it), and just disable consul? It's easier to see diff between generated resources that way.

Reply:
Done, for convenience the diff is
```diff
--- test-gossip-generated.yaml	2022-06-14 09:28:00.000000000 -0400
+++ test-gossip-disable-consul-generated.yaml	2022-06-14 09:27:59.000000000 -0400
@@ -43,225 +43,6 @@
       name: store-gateway
 ---
 apiVersion: v1
-kind: ServiceAccount
-metadata:
-  name: consul-sidekick
-  namespace: default
----
-apiVersion: v1
-data:
-  consul-config.json: '{"leave_on_terminate": true, "raft_snapshot_threshold": 128,
-    "raft_trailing_logs": 10000, "telemetry": {"dogstatsd_addr": "127.0.0.1:9125"}}'
-  mapping: |
-    mappings:
-    - match: consul.*.runtime.*
-      name: consul_runtime
-      labels:
-        type: $2
-    - match: consul.runtime.total_gc_pause_ns
-      name: consul_runtime_total_gc_pause_ns
-      labels:
-        type: $2
-    - match: consul.consul.health.service.query-tag.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3
-    - match: consul.consul.health.service.query-tag.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5.$6
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5.$6.$7
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5.$6.$7.$8
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5.$6.$7.$8.$9
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5.$6.$7.$8.$9.$10
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*.*.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5.$6.$7.$8.$9.$10.$11
-    - match: consul.consul.health.service.query-tag.*.*.*.*.*.*.*.*.*.*.*.*
-      name: consul_health_service_query_tag
-      labels:
-        query: $1.$2.$3.$4.$5.$6.$7.$8.$9.$10.$11.$12
-    - match: consul.consul.catalog.deregister
-      name: consul_catalog_deregister
-      labels: {}
-    - match: consul.consul.dns.domain_query.*.*.*.*.*
-      name: consul_dns_domain_query
-      labels:
-        query: $1.$2.$3.$4.$5
-    - match: consul.consul.health.service.not-found.*
-      name: consul_health_service_not_found
-      labels:
-        query: $1
-    - match: consul.consul.health.service.query.*
-      name: consul_health_service_query
-      labels:
-        query: $1
-    - match: consul.*.memberlist.health.score
-      name: consul_memberlist_health_score
-      labels: {}
-    - match: consul.serf.queue.*
-      name: consul_serf_events
-      labels:
-        type: $1
-    - match: consul.serf.snapshot.appendLine
-      name: consul_serf_snapshot_appendLine
-      labels:
-        type: $1
-    - match: consul.serf.coordinate.adjustment-ms
-      name: consul_serf_coordinate_adjustment_ms
-      labels: {}
-    - match: consul.consul.rpc.query
-      name: consul_rpc_query
-      labels: {}
-    - match: consul.*.consul.session_ttl.active
-      name: consul_session_ttl_active
-      labels: {}
-    - match: consul.raft.rpc.*
-      name: consul_raft_rpc
-      labels:
-        type: $1
-    - match: consul.raft.rpc.appendEntries.storeLogs
-      name: consul_raft_rpc_appendEntries_storeLogs
-      labels:
-        type: $1
-    - match: consul.consul.fsm.persist
-      name: consul_fsm_persist
-      labels: {}
-    - match: consul.raft.fsm.apply
-      name: consul_raft_fsm_apply
-      labels: {}
-    - match: consul.raft.leader.lastContact
-      name: consul_raft_leader_lastcontact
-      labels: {}
-    - match: consul.raft.leader.dispatchLog
-      name: consul_raft_leader_dispatchLog
-      labels: {}
-    - match: consul.raft.commitTime
-      name: consul_raft_commitTime
-      labels: {}
-    - match: consul.raft.replication.appendEntries.logs.*.*.*.*
-      name: consul_raft_replication_appendEntries_logs
-      labels:
-        query: ${1}.${2}.${3}.${4}
-    - match: consul.raft.replication.appendEntries.rpc.*.*.*.*
-      name: consul_raft_replication_appendEntries_rpc
-      labels:
-        query: ${1}.${2}.${3}.${4}
-    - match: consul.raft.replication.heartbeat.*.*.*.*
-      name: consul_raft_replication_heartbeat
-      labels:
-        query: ${1}.${2}.${3}.${4}
-    - match: consul.consul.rpc.request
-      name: consul_rpc_requests
-      labels: {}
-    - match: consul.consul.rpc.accept_conn
-      name: consul_rpc_accept_conn
-      labels: {}
-    - match: consul.memberlist.udp.*
-      name: consul_memberlist_udp
-      labels:
-        type: $1
-    - match: consul.memberlist.tcp.*
-      name: consul_memberlist_tcp
-      labels:
-        type: $1
-    - match: consul.memberlist.gossip
-      name: consul_memberlist_gossip
-      labels: {}
-    - match: consul.memberlist.probeNode
-      name: consul_memberlist_probenode
-      labels: {}
-    - match: consul.memberlist.pushPullNode
-      name: consul_memberlist_pushpullnode
-      labels: {}
-    - match: consul.http.*
-      name: consul_http_request
-      labels:
-        method: $1
-        path: /
-    - match: consul.http.*.*
-      name: consul_http_request
-      labels:
-        method: $1
-        path: /$2
-    - match: consul.http.*.*.*
-      name: consul_http_request
-      labels:
-        method: $1
-        path: /$2/$3
-    - match: consul.http.*.*.*.*
-      name: consul_http_request
-      labels:
-        method: $1
-        path: /$2/$3/$4
-    - match: consul.http.*.*.*.*.*
-      name: consul_http_request
-      labels:
-        method: $1
-        path: /$2/$3/$4/$5
-    - match: consul.consul.leader.barrier
-      name: consul_leader_barrier
-      labels: {}
-    - match: consul.consul.leader.reconcileMember
-      name: consul_leader_reconcileMember
-      labels: {}
-    - match: consul.consul.leader.reconcile
-      name: consul_leader_reconcile
-      labels: {}
-    - match: consul.consul.fsm.coordinate.batch-update
-      name: consul_fsm_coordinate_batch_update
-      labels: {}
-    - match: consul.consul.fsm.autopilot
-      name: consul_fsm_autopilot
-      labels: {}
-    - match: consul.consul.fsm.kvs.cas
-      name: consul_fsm_kvs_cas
-      labels: {}
-    - match: consul.consul.fsm.register
-      name: consul_fsm_register
-      labels: {}
-    - match: consul.consul.fsm.deregister
-      name: consul_fsm_deregister
-      labels: {}
-    - match: consul.consul.fsm.tombstone.reap
-      name: consul_fsm_tombstone_reap
-      labels: {}
-    - match: consul.consul.catalog.register
-      name: consul_catalog_register
-      labels: {}
-    - match: consul.consul.catalog.deregister
-      name: consul_catalog_deregister
-      labels: {}
-    - match: consul.consul.leader.reapTombstones
-      name: consul_leader_reapTombstones
-      labels: {}
-kind: ConfigMap
-metadata:
-  name: consul
-  namespace: default
----
-apiVersion: v1
 data:
   overrides.yaml: |
     overrides: {}
@@ -270,38 +51,6 @@
   name: overrides
   namespace: default
 ---
-apiVersion: rbac.authorization.k8s.io/v1
-kind: Role
-metadata:
-  name: consul-sidekick
-  namespace: default
-rules:
-- apiGroups:
-  - ""
-  - extensions
-  - apps
-  resources:
-  - pods
-  - replicasets
-  verbs:
-  - get
-  - list
-  - watch
----
-apiVersion: rbac.authorization.k8s.io/v1
-kind: RoleBinding
-metadata:
-  name: consul-sidekick
-  namespace: default
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: Role
-  name: consul-sidekick
-subjects:
-- kind: ServiceAccount
-  name: consul-sidekick
-  namespace: default
----
 apiVersion: v1
 kind: Service
 metadata:
@@ -347,36 +96,6 @@
 kind: Service
 metadata:
   labels:
-    name: consul
-  name: consul
-  namespace: default
-spec:
-  ports:
-  - name: consul-server
-    port: 8300
-    targetPort: 8300
-  - name: consul-serf
-    port: 8301
-    targetPort: 8301
-  - name: consul-client
-    port: 8400
-    targetPort: 8400
-  - name: consul-api
-    port: 8500
-    targetPort: 8500
-  - name: statsd-exporter-http-metrics
-    port: 8000
-    targetPort: 8000
-  - name: consul-exporter-http-metrics
-    port: 9107
-    targetPort: 9107
-  selector:
-    name: consul
----
-apiVersion: v1
-kind: Service
-metadata:
-  labels:
     name: distributor
   name: distributor
   namespace: default
@@ -646,131 +365,6 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: consul
-  namespace: default
-spec:
-  minReadySeconds: 10
-  replicas: 1
-  revisionHistoryLimit: 10
-  selector:
-    matchLabels:
-      name: consul
-  template:
-    metadata:
-      annotations:
-        consul-hash: e56ef6821a3557604caccaf6d5820239
-      labels:
-        name: consul
-    spec:
-      affinity:
-        podAntiAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-          - labelSelector:
-              matchLabels:
-                name: consul
-            topologyKey: kubernetes.io/hostname
-          - labelSelector:
-              matchLabels:
-                name: ingester
-            namespaces:
-            - default
-            topologyKey: kubernetes.io/hostname
-      containers:
-      - args:
-        - agent
-        - -ui
-        - -server
-        - -client=0.0.0.0
-        - -config-file=/etc/config/consul-config.json
-        - -bootstrap-expect=1
-        - -ui-content-path=/default/consul/
-        env:
-        - name: CHECKPOINT_DISABLE
-          value: "1"
-        image: consul:1.5.3
-        imagePullPolicy: IfNotPresent
-        name: consul
-        ports:
-        - containerPort: 8300
-          name: server
-        - containerPort: 8301
-          name: serf
-        - containerPort: 8400
-          name: client
-        - containerPort: 8500
-          name: api
-        resources:
-          requests:
-            cpu: "4"
-            memory: 4Gi
-        volumeMounts:
-        - mountPath: /etc/config
-          name: consul
-        - mountPath: /consul/data/
-          name: data
-      - args:
-        - --namespace=$(POD_NAMESPACE)
-        - --pod-name=$(POD_NAME)
-        env:
-        - name: POD_NAMESPACE
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.namespace
-        - name: POD_NAME
-          valueFrom:
-            fieldRef:
-              fieldPath: metadata.name
-        image: weaveworks/consul-sidekick:master-f18ad13
-        imagePullPolicy: IfNotPresent
-        name: sidekick
-        volumeMounts:
-        - mountPath: /etc/config
-          name: consul
-        - mountPath: /consul/data/
-          name: data
-      - args:
-        - --web.listen-address=:8000
-        - --statsd.mapping-config=/etc/config/mapping
-        image: prom/statsd-exporter:v0.12.2
-        imagePullPolicy: IfNotPresent
-        name: statsd-exporter
-        ports:
-        - containerPort: 8000
-          name: http-metrics
-        volumeMounts:
-        - mountPath: /etc/config
-          name: consul
-        - mountPath: /consul/data/
-          name: data
-      - args:
-        - --consul.server=localhost:8500
-        - --web.listen-address=:9107
-        - --consul.timeout=1s
-        - --no-consul.health-summary
-        - --consul.allow_stale
-        image: prom/consul-exporter:v0.5.0
-        imagePullPolicy: IfNotPresent
-        name: consul-exporter
-        ports:
-        - containerPort: 9107
-          name: http-metrics
-        volumeMounts:
-        - mountPath: /etc/config
-          name: consul
-        - mountPath: /consul/data/
-          name: data
-      serviceAccount: consul-sidekick
-      volumes:
-      - configMap:
-          name: consul
-        name: consul
-      - emptyDir:
-          medium: Memory
-        name: data
----
-apiVersion: apps/v1
-kind: Deployment
-metadata:
   name: distributor
   namespace: default
 spec:
```

Reply:
Funny thing is that there is still `consul` reference in `test-gossip-disable-consul-generated.yaml`, used by alertmananager:

```
        - -alertmanager.sharding-ring.consul.hostname=consul.default.svc.cluster.local:8500
        - -alertmanager.sharding-ring.replication-factor=3
        - -alertmanager.sharding-ring.store=consul
```

Alertmanager should respect `memberlist_enabled` flag as well. We should fix that (in separate PR).

Reply:
Opened https://github.com/grafana/mimir/pull/2102.